scripts/python-skywater-pdk/skywater_pdk/gds_to_svg/gds_to_svg.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 SkyWater PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+import re
+import subprocess
+import argparse
+
+FATAL_ERROR = re.compile('((Error parsing)|(No such file or directory)|(couldn\'t be read))')  # noqa: E501
+READING_REGEX = re.compile('Reading "([^"]*)".')
+
+debug = True
+superdebug = True
+
+
+def _magic_tcl_header(ofile, gdsfile):
+    """
+    Adds a header to TCL file.
+
+    Parameters
+    ----------
+    ofile: output file stream
+    gdsfile: path to GDS file
+    """
+    print('#!/bin/env wish',         file=ofile)
+    print('drc off',                 file=ofile)
+    print('scalegrid 1 2',           file=ofile)
+    print('cif istyle vendorimport', file=ofile)
+    print('gds readonly true',       file=ofile)
+    print('gds rescale false',       file=ofile)
+    print('tech unlock *',           file=ofile)
+    print('cif warning default',     file=ofile)
+    print('set VDD VPWR',            file=ofile)
+    print('set GND VGND',            file=ofile)
+    print('set SUB SUBS',            file=ofile)
+    print('gds read ' + gdsfile,     file=ofile)
+
+
+def run_magic(destdir, tcl_path, input_techfile, d="null"):
+    """
+    Runs magic to generate layout files.
+
+    Parameters
+    ----------
+    destdir: destination directory
+    tcl_path: path to input TCL file
+    input_techfile: path to the technology file
+    d: Workstation type, can be NULL, X11, OGL or XWIND
+    """
+    cmd = [
+        'magic',
+        '-nowrapper',
+        '-d'+d,
+        '-noconsole',
+        '-T', input_techfile,
+        os.path.abspath(tcl_path)
+    ]
+    with open(tcl_path.replace(".tcl", ".sh"), "w") as f:
+        f.write("#!/bin/sh\n")
+        f.write(" ".join(cmd))
+    mproc = subprocess.run(
+        cmd,
+        stdin=subprocess.DEVNULL,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        cwd=destdir,
+        universal_newlines=True)
+    assert mproc.stdout
+    max_cellname_width = 0
+    output_by_cells = [('', [])]
```

Comment:
The output below could use some comments / cleanup.

I would also suggest the magic functions end up in a file like `skywater_pdk/tools/magic.py` so they can be reused by other generators?

Reply:
I cleaned up the code and outputs, improved the docs for outputs, moved explicitly `magic`-related stuff to `skywater_pdk.tools.magic`, and drawing-related stuff to `skywater_pdk.tools.draw` (inkscape calls).

Reply:
I also moved `skywater_pdk/gds_to_svg/gds_to_svg.py` to `skywater_pdk/gds_to_svg.py` and `skywater_pdk/gds_to_svg/generate-gds-svgs.py` to `skywater_pdk/cell/generate/layout.py`.